src/main/java/org/opensearch/knn/index/query/ExactSearcher.java

```diff
@@ -87,7 +92,10 @@ private Map<Integer, Float> doRadialSearch(
     ) throws IOException {
         final SegmentReader reader = Lucene.segmentReader(leafReaderContext.reader());
         final KNNQuery knnQuery = exactSearcherContext.getKnnQuery();
-        final FieldInfo fieldInfo = reader.getFieldInfos().fieldInfo(knnQuery.getField());
+        final FieldInfo fieldInfo = FieldInfoExtractor.getFieldInfo(reader, knnQuery.getField());
+        if (fieldInfo == null) {
```

Comment:
Worth adding a case in unit test

Reply:
sure for k-NN search I have added the test but its not there for radial search. Will add it